src/wp-includes/block-editor.php

```diff
@@ -265,3 +265,71 @@ function get_block_editor_settings( $editor_name, $custom_settings = array() ) {
 
 	return $editor_settings;
 }
+
+/**
+ * Preloads common data used with the block editor by specifying an array of
+ * REST API paths that will be preloaded for a given editor type.
+ *
+ * @since 5.8.0
+ *
+ * @param string $editor_name   The name of the editor (e.g. 'post-editor').
+ * @param array  $preload_paths List of paths to preload.
+ *
+ * @return void
+ */
+function block_editor_preload_api_requests( $editor_name, $preload_paths ) {
+	/**
+	 * Filters the array of REST API paths that will be used to preloaded common data
+	 * to use with the block editor.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param string[] $preload_paths Array of paths to preload.
+	 */
+	$preload_paths = apply_filters( "block_editor_preload_paths_{$editor_name}", $preload_paths );
+	if ( 'post-editor' === $editor_name ) {
+		$post = get_post();
+
+		/**
+		 * Preload common data by specifying an array of REST API paths that will be preloaded.
+		 *
+		 * Filters the array of paths that will be preloaded.
+		 *
+		 * @since 5.0.0
+		 * @deprecated 5.8.0 The hook transitioned to support also screens that don't contain $post instance.
+		 *
+		 * @param string[] $preload_paths Array of paths to preload.
+		 * @param WP_Post  $post          Post being edited.
+		 */
+		$preload_paths = apply_filters_deprecated( 'block_editor_preload_paths', array( $preload_paths, $post ), '5.8.0', "block_editor_preload_paths_{$editor_name}" );
+	}
+
+	if ( empty( $preload_paths ) ) {
+		return;
+	}
+
+	/*
+	 * Ensure the global $post remains the same after API data is preloaded.
+	 * Because API preloading can call the_content and other filters, plugins
+	 * can unexpectedly modify $post.
+	 */
+	$backup_global_post = ! empty( $post ) ? clone $post : $post;
```

Comment:
`$post` is no longer defined here. It should use global or `get_post` outside of the if statement.